iroh-store/src/rpc.rs

```diff
@@ -18,107 +16,131 @@ impl iroh_rpc_types::NamedService for Store {
+            .spawn_blocking(move |x| {
+                if let Some(res) = x.get(&cid)? {
+                    Ok(GetResponse {
+                        data: Some(BytesMut::from(&res[..]).freeze()),
+                    })
+                } else {
+                    Ok(GetResponse { data: None })
+                }
+            })
+            .await
     }
 
     #[tracing::instrument(skip(self))]
-    async fn has(&self, req: HasRequest) -> Result<HasResponse> {
-        let cid = cid_from_bytes(req.cid)?;
-        self.spawn_blocking(move |self| {
-            let has = self.has(&cid)?;
-            Ok(HasResponse { has })
-        })
-        .await
+    async fn has(self, req: HasRequest) -> Result<HasResponse> {
+        let cid = req.cid;
+        self.0
+            .spawn_blocking(move |x| {
+                let has = x.has(&cid)?;
+                Ok(HasResponse { has })
+            })
+            .await
     }
 
     #[tracing::instrument(skip(self))]
-    async fn get_links(&self, req: GetLinksRequest) -> Result<GetLinksResponse> {
-        let cid = cid_from_bytes(req.cid)?;
-        self.spawn_blocking(move |self| {
-            if let Some(res) = self.get_links(&cid)? {
-                let links = res.into_iter().map(|cid| cid.to_bytes()).collect();
+    async fn get_links(self, req: GetLinksRequest) -> Result<GetLinksResponse> {
+        let cid = req.cid;
+        self.0
+            .spawn_blocking(move |x| {
+                let links = x.get_links(&cid)?;
                 Ok(GetLinksResponse { links })
-            } else {
-                Ok(GetLinksResponse { links: Vec::new() })
-            }
-        })
-        .await
+            })
+            .await
     }
 
     #[tracing::instrument(skip(self))]
-    async fn get_size(&self, req: GetSizeRequest) -> Result<GetSizeResponse> {
-        let cid = cid_from_bytes(req.cid)?;
-        self.spawn_blocking(move |self| {
-            if let Some(size) = self.get_size(&cid)? {
-                Ok(GetSizeResponse {
-                    size: Some(size as u64),
-                })
-            } else {
-                Ok(GetSizeResponse { size: None })
+    async fn get_size(self, req: GetSizeRequest) -> Result<GetSizeResponse> {
+        let cid = req.cid;
+        self.0
+            .spawn_blocking(move |x| {
+                let size = x.get_size(&cid)?.map(|x| x as u64);
+                Ok(GetSizeResponse { size })
+            })
+            .await
+    }
+}
+
+/// Handle a session with a client. This will loop until either the client closes the connection or
+/// one of the requests produces an error.
+async fn handle_session(server: StoreServer, store: Store) -> Result<()> {
+    use StoreRequest::*;
+    let s = server.clone();
+    let store = RpcStore(store);
+    loop {
+        let store = store.clone();
+        let s = s.clone();
+        let (req, chan) = s.accept_one().await?;
+        let store = store.clone();
+        #[rustfmt::skip]
```

Comment:
why?

Reply:
Because I don't want the long lines to be broken up.

Note that the giant switch statement is also something that the macro can do for you, if you don't like it.

But I think that having it in the open is not that bad. E.g. you can add custom timeouts to individual RPC calls by just wrapping the call, whereas with a macro you would need to come up with some macro arguments etc.